sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,42 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        private BinaryData? _content;
+
+        /// <summary>
+        /// Gets the contents of HTTP response. Returns a BinaryData with no payload for responses without content.
+        /// </summary>
+        public virtual BinaryData Content
+        {
+            get
+            {
+                if (_content != null)
+                {
+                    return _content;
```

Comment:
The `ContentStream` property of `Response` can be `set` (but I'm not really sure in what cases that ends up happening).  For our implementations, would we want to throw away this cached value in the case where the stream is replaced (so if after replacing the stream you call `.Content` you see the new content)?

Reply:
I think we have no choice but to throw it away. But then how do we read the stream in a sync context (if it's not a memory stream). I guess we just sync over async. This is a bit of a mess.